### PR TITLE
Elevated tokens

### PR DESCRIPTION
**Pull request by joshgummersall**
Refactor logic to handlers, begin email handler.

### CODE REVIEW

File: util/http.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+func ReplyJSON(w http.ResponseWriter, code int, r interface{}) {
+	b, err := json.Marshal(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	w.WriteHeader(code)
```
Comment (wescarr) [inline util/http.go:21]: I assume this is the http status code?

File: serve/main.go [hunk 2]
```diff
@@ -1,326 +1,59 @@
...
 
-	tokenIssuer = jwt.NewIssuer(
-		[]byte(getEnvVar("JWT_SIGNING_KEY")),
-		fmt.Sprintf("%s/operator", tokenNamespace),
-		fmt.Sprintf("%s/token", tokenNamespace),
-		td,
-	)
-
-	rd, err := time.ParseDuration(getEnvVar("JWT_REFRESH_DURATION"))
+	rd, err := time.ParseDuration(util.GetEnvVar("JWT_REFRESH_DURATION"))
```
Comment (wescarr) [inline serve/main.go:30]: Does it make sense that both elevated and non-elevated JWTs have the same refresh duration?

File: handler/email.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	return h.bucket
+}
+
+type emailReq struct {
+	Email string `json:"email"`
+}
+
+var emailKey contextKey
+
+func (h *emailHandler) Meta(ctx context.Context, r *http.Request) (context.Context, *string, *tokenMeta, error) {
```
Comment (wescarr) [inline handler/email.go:63]: The `Meta` name for this function is a bit odd IMO. I would either add a comment to the Handler interface that describes what this does, or rename it to something more descriptive.

**Review by wescarr — COMMENTED** [review]
Looks good overall!

File: util/http.go (hunk 1 above)
Comment (joshgummersall) [reply util/http.go:21]: Yup

File: serve/main.go (hunk 2 above)
Comment (joshgummersall) [reply serve/main.go:30]: I think it's probably fine but I'll make them separate env vars.

File: handler/email.go (hunk 3 above)
Comment (joshgummersall) [reply handler/email.go]: Renamed, and simplified the gnarly context stuff by shoving a few more things in the `tokenMeta` struct.

**Comment by joshgummersall** [comment]
TODO tomorrow:

- [ ] Email whitelist
- [ ] Email service integration
- [ ] Shove email into JWT, probably into `aud` claim

**Comment by joshgummersall** [comment]
@wes the JWT changes are: https://github.com/covidtrace/jwt/pull/1

**Comment by joshgummersall** [comment]
tf changes https://github.com/covidtrace/infra/pull/3

**Review by wescarr — APPROVED** [review]
Thanks for doing this! I'll talk to @dudleycarr about which email provider to use today

**Pull request merged**